alerting/src/main/kotlin/org/opensearch/alerting/action/DocLevelMonitorFanOutRequest.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.alerting.action
+
+import org.opensearch.action.ActionRequest
+import org.opensearch.action.ActionRequestValidationException
+import org.opensearch.alerting.model.IndexExecutionContext
+import org.opensearch.alerting.model.MonitorMetadata
+import org.opensearch.alerting.workflow.WorkflowRunContext
+import org.opensearch.commons.alerting.model.Monitor
+import org.opensearch.core.common.io.stream.StreamInput
+import org.opensearch.core.common.io.stream.StreamOutput
+import org.opensearch.core.index.shard.ShardId
+import org.opensearch.core.xcontent.ToXContent
+import org.opensearch.core.xcontent.ToXContentObject
+import org.opensearch.core.xcontent.XContentBuilder
+import java.io.IOException
+
+class DocLevelMonitorFanOutRequest : ActionRequest, ToXContentObject {
```

Comment:
plz add ser-deser tests for writeTo(), readFRom(), xcontent roudntrip(), override equals() method (or whatever is the kotlin equivalent of deciding how objects are deemed equal)

Reply:
ser-deser tests added. no need for `xcontent roudntrip()` as there is no rest layer for these objects.